<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_17) on Fri Feb 14 19:40:41 PST 2014 -->
<TITLE>
ExPANdS
</TITLE>

<META NAME="date" CONTENT="2014-02-14">

<LINK REL ="stylesheet" TYPE="text/css" HREF="stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="ExPANdS";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ExPANdS.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV CLASS&nbsp;
&nbsp;<A HREF="ExPANdS.FitAlternatives.html" title="class in &lt;Unnamed&gt;"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="index.html?ExPANdS.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ExPANdS.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
Class ExPANdS</H2>
<PRE>
java.lang.Object
  <IMG SRC="./resources/inherit.gif" ALT="extended by "><B>ExPANdS</B>
</PRE>
<HR>
<DL>
<DT><PRE>public class <B>ExPANdS</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
Class used by the R-package EXPANDS (Expanding Ploidy and Allele Frequency on
 Nested Subpopulations), that characterizes coexisting SPs in a tumor using
 copy number and B-allele frequencies derived from exome- or whole genome
 sequencing input data. Functionality provides exhaustive iteration through
 scenarios that explain the B-allele frequency and copy number measured for a
 given locus, while considering cell type specific ploidies, cellular
 frequencies and noise in copy number measurements . Functionalities concern
 individual point mutations only, rather than mutation clusters.
<P>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>Noemi</DD>
</DL>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->

<A NAME="nested_class_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Nested Class Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.FitAlternatives.html" title="class in &lt;Unnamed&gt;">ExPANdS.FitAlternatives</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Stores alternative cellular frequencies f, that fit the equations:
 PM*f+PN*(1-f)=CN and PM_B*f+PN_B*(1-f)=CN*AF, where CN and AF are copy
 number and B-allele frequency measurements derived from NGS, while (PM, PN,
 PM_B, PN_B)=(total ploidy in mutated cells, total ploidy in normal cells,
 B-allele ploidy in mutated cells, B-allele ploidy in normal cells) are
 fixed.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.Solution.html" title="class in &lt;Unnamed&gt;">ExPANdS.Solution</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Stores alternative solutions (PM_B, PM, f)=(B-allele ploidy, total ploidy,
 cellular frequency) computed for the NGS derived measures: AF - B-allele
 frequency, CN - copy number and PN_B - ploidy of the B-allele in normal
 cells, that are associated with a mutated locus.</TD>
</TR>
</TABLE>
&nbsp;<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>private &nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#AF">AF</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Observed/measured B-allele frequency</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>private &nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#alternativeCN">alternativeCN</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Copy number alternatives obtained by adding different levels of noise to
 the measured copy number: alternativeCN:=(CN - maxCN_Error):(CN +
 maxCN_Error), where CN is the measured copy number and maxCN_error is the
 maximal level of noise.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>private &nbsp;<A HREF="ExPANdS.FitAlternatives.html" title="class in &lt;Unnamed&gt;">ExPANdS.FitAlternatives</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#bestSolution">bestSolution</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Cellular frequency alternatives for fixed PM and PM_B</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>private &nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#CN">CN</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Observed/measured copy number</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#DEFAULT_MAX_PM">DEFAULT_MAX_PM</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Stores default value of maximal ploidy in mutated cells, accepted as
 solution.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>private &nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#max_PM">max_PM</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Maximal ploidy in mutated cells, accepted as solution.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>private static&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#maxCN_Error">maxCN_Error</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Maximal value for estimated copy-number error (default 0.1)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>private static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#min_PM">min_PM</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Minimal ploidy in mutated cells (PM=0 is not accepted, since B-allele
 ploidy equation not applicable)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>private &nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#minErr">minErr</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The error associated with the currently best solution</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>private static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#PN">PN</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Total ploidy in normal cells</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>private &nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#PN_B">PN_B</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Observed/measured ploidy of B-allele in normal cells</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#SOLUTION_ENTITIES">SOLUTION_ENTITIES</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;To be used as column headers - components constituting the inputs and
 outputs of a solution</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>private &nbsp;java.util.List&lt;<A HREF="ExPANdS.FitAlternatives.html" title="class in &lt;Unnamed&gt;">ExPANdS.FitAlternatives</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#solutions">solutions</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Alternative solutions, consisting of (PM, PM_B, f).</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="ExPANdS.html#ExPANdS(double, double, int, int)">ExPANdS</A></B>(double&nbsp;af,
        double&nbsp;cn,
        int&nbsp;pnb,
        int&nbsp;max_PM)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates alternative solutions, consisting of (PM, PM_B, f) =(total
 ploidy in mutated cells, B-allele ploidy in mutated cells, frequency of
 mutated cells), that can explain B-allele frequency and copy number
 measured for a single mutated locus.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#getDeviation()">getDeviation</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Deviation between observed and expected values for B-allele frequency and
 copy number.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#getF()">getF</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Frequency of cells harboring the mutation at this locus, obtained by
 minimizing the deviation between observed and expected B-allele frequency
 and copy number.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#getPM_B()">getPM_B</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;B-allele ploidy of this locus, obtained by minimizing the deviation between
 observed and expected B-allele frequency and copy number.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#getPM()">getPM</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Total ploidy of this locus, obtained by minimizing the deviation between
 observed and expected B-allele frequency and copy number.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#printFitsToFile(java.io.File)">printFitsToFile</A></B>(java.io.File&nbsp;f)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#run()">run</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Exhaustive iteration through all possible combinations of (PM,PM_B,f),
 while considering copy number noise</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;<A HREF="ExPANdS.FitAlternatives.html" title="class in &lt;Unnamed&gt;">ExPANdS.FitAlternatives</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="ExPANdS.html#solutions()">solutions</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Alternative solutions (PM_B, PM, f)=(B-allele ploidy, total ploidy,
 cellular frequency) computed to explain NGS derived allele frequency and
 copy number measurements.</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="DEFAULT_MAX_PM"><!-- --></A><H3>
DEFAULT_MAX_PM</H3>
<PRE>
public static final int <B>DEFAULT_MAX_PM</B></PRE>
<DL>
<DD>Stores default value of maximal ploidy in mutated cells, accepted as
 solution.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="constant-values.html#ExPANdS.DEFAULT_MAX_PM">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="min_PM"><!-- --></A><H3>
min_PM</H3>
<PRE>
private static int <B>min_PM</B></PRE>
<DL>
<DD>Minimal ploidy in mutated cells (PM=0 is not accepted, since B-allele
 ploidy equation not applicable)
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="PN"><!-- --></A><H3>
PN</H3>
<PRE>
private static int <B>PN</B></PRE>
<DL>
<DD>Total ploidy in normal cells
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="solutions"><!-- --></A><H3>
solutions</H3>
<PRE>
private java.util.List&lt;<A HREF="ExPANdS.FitAlternatives.html" title="class in &lt;Unnamed&gt;">ExPANdS.FitAlternatives</A>&gt; <B>solutions</B></PRE>
<DL>
<DD>Alternative solutions, consisting of (PM, PM_B, f). Each entry in the list
 consists of fixed values for PM and PM_B and variable f (frequency of
 mutated cell)
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="CN"><!-- --></A><H3>
CN</H3>
<PRE>
private double <B>CN</B></PRE>
<DL>
<DD>Observed/measured copy number
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="AF"><!-- --></A><H3>
AF</H3>
<PRE>
private double <B>AF</B></PRE>
<DL>
<DD>Observed/measured B-allele frequency
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="maxCN_Error"><!-- --></A><H3>
maxCN_Error</H3>
<PRE>
private static double <B>maxCN_Error</B></PRE>
<DL>
<DD>Maximal value for estimated copy-number error (default 0.1)
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="PN_B"><!-- --></A><H3>
PN_B</H3>
<PRE>
private int <B>PN_B</B></PRE>
<DL>
<DD>Observed/measured ploidy of B-allele in normal cells
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="minErr"><!-- --></A><H3>
minErr</H3>
<PRE>
private double <B>minErr</B></PRE>
<DL>
<DD>The error associated with the currently best solution
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="alternativeCN"><!-- --></A><H3>
alternativeCN</H3>
<PRE>
private double[] <B>alternativeCN</B></PRE>
<DL>
<DD>Copy number alternatives obtained by adding different levels of noise to
 the measured copy number: alternativeCN:=(CN - maxCN_Error):(CN +
 maxCN_Error), where CN is the measured copy number and maxCN_error is the
 maximal level of noise.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="bestSolution"><!-- --></A><H3>
bestSolution</H3>
<PRE>
private <A HREF="ExPANdS.FitAlternatives.html" title="class in &lt;Unnamed&gt;">ExPANdS.FitAlternatives</A> <B>bestSolution</B></PRE>
<DL>
<DD>Cellular frequency alternatives for fixed PM and PM_B
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="max_PM"><!-- --></A><H3>
max_PM</H3>
<PRE>
private int <B>max_PM</B></PRE>
<DL>
<DD>Maximal ploidy in mutated cells, accepted as solution.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="SOLUTION_ENTITIES"><!-- --></A><H3>
SOLUTION_ENTITIES</H3>
<PRE>
public static final java.lang.String[] <B>SOLUTION_ENTITIES</B></PRE>
<DL>
<DD>To be used as column headers - components constituting the inputs and
 outputs of a solution
<P>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="ExPANdS(double, double, int, int)"><!-- --></A><H3>
ExPANdS</H3>
<PRE>
public <B>ExPANdS</B>(double&nbsp;af,
               double&nbsp;cn,
               int&nbsp;pnb,
               int&nbsp;max_PM)</PRE>
<DL>
<DD>Calculates alternative solutions, consisting of (PM, PM_B, f) =(total
 ploidy in mutated cells, B-allele ploidy in mutated cells, frequency of
 mutated cells), that can explain B-allele frequency and copy number
 measured for a single mutated locus.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>af</CODE> - measured B-allele frequency<DD><CODE>cn</CODE> - measured copy number<DD><CODE>pnb</CODE> - ploidy of the B-allel in normal cells<DD><CODE>max_PM</CODE> - maximal total ploidy in mutated cells</DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getDeviation()"><!-- --></A><H3>
getDeviation</H3>
<PRE>
public double <B>getDeviation</B>()</PRE>
<DL>
<DD>Deviation between observed and expected values for B-allele frequency and
 copy number. The value of this function is 0, if and only if the components
 of the best solution yield exactly the same copy number and allele
 frequency as measured.
<P>
<DD><DL>

<DT><B>Returns:</B><DD>the deviation of the observed from the expected copy number and
         allele frequency, for the inferred solution</DL>
</DD>
</DL>
<HR>

<A NAME="run()"><!-- --></A><H3>
run</H3>
<PRE>
public void <B>run</B>()</PRE>
<DL>
<DD>Exhaustive iteration through all possible combinations of (PM,PM_B,f),
 while considering copy number noise
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getPM_B()"><!-- --></A><H3>
getPM_B</H3>
<PRE>
public int <B>getPM_B</B>()</PRE>
<DL>
<DD>B-allele ploidy of this locus, obtained by minimizing the deviation between
 observed and expected B-allele frequency and copy number.
<P>
<DD><DL>

<DT><B>Returns:</B><DD>ploidy of the B-allele</DL>
</DD>
</DL>
<HR>

<A NAME="getPM()"><!-- --></A><H3>
getPM</H3>
<PRE>
public int <B>getPM</B>()</PRE>
<DL>
<DD>Total ploidy of this locus, obtained by minimizing the deviation between
 observed and expected B-allele frequency and copy number.
<P>
<DD><DL>

<DT><B>Returns:</B><DD>total ploidy</DL>
</DD>
</DL>
<HR>

<A NAME="getF()"><!-- --></A><H3>
getF</H3>
<PRE>
public double <B>getF</B>()</PRE>
<DL>
<DD>Frequency of cells harboring the mutation at this locus, obtained by
 minimizing the deviation between observed and expected B-allele frequency
 and copy number.
<P>
<DD><DL>

<DT><B>Returns:</B><DD>cellular frequency</DL>
</DD>
</DL>
<HR>

<A NAME="printFitsToFile(java.io.File)"><!-- --></A><H3>
printFitsToFile</H3>
<PRE>
public void <B>printFitsToFile</B>(java.io.File&nbsp;f)
                     throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="solutions()"><!-- --></A><H3>
solutions</H3>
<PRE>
public java.util.Collection&lt;<A HREF="ExPANdS.FitAlternatives.html" title="class in &lt;Unnamed&gt;">ExPANdS.FitAlternatives</A>&gt; <B>solutions</B>()</PRE>
<DL>
<DD>Alternative solutions (PM_B, PM, f)=(B-allele ploidy, total ploidy,
 cellular frequency) computed to explain NGS derived allele frequency and
 copy number measurements.
<P>
<DD><DL>

<DT><B>Returns:</B><DD>List of alternative solutions</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ExPANdS.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV CLASS&nbsp;
&nbsp;<A HREF="ExPANdS.FitAlternatives.html" title="class in &lt;Unnamed&gt;"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="index.html?ExPANdS.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ExPANdS.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
